### ISSUE TITLE
Uploading 4 CSVs (by accident) crashes Explorer.exe

### ISSUE THREAD

**Issue by gaia** [ShareX/ShareX#1948]
Debug Log: http://pastebin.com/Y6nss178

Win 10 x64 Build 1607

I right clicked in explorer on 4 CSV files and accidentally chose the ShareX option instead of sublime text. it crashed Explorer:

```
Faulting application name: Explorer.EXE, version: 10.0.14393.0, time stamp: 0x57899981
Faulting module name: ntdll.dll, version: 10.0.14393.103, time stamp: 0x57b7e207
Exception code: 0xc0000374
Fault offset: 0x00000000000f73f3
Faulting process id: 0x1cb0
Faulting application start time: 0x01d218051fc3ae47
Faulting application path: C:\WINDOWS\Explorer.EXE
Faulting module path: C:\WINDOWS\SYSTEM32\ntdll.dll
Report Id: d07ebbbf-fa53-4790-
Faulting package full name: 
Faulting package-relative application ID:
```

followed by

```

ShellExperienceHost.exe
10.0.14393.187
57cf9d73
Windows.UI.Xaml.dll
10.0.14393.187
57cf9d04
c000027b
00000000006d638b
20ac
01d21805xxxxxxx
C:\WINDOWS\SystemApps\ShellExperienceHost_cw5n1h2txyewy\ShellExperienceHost.exe
C:\Windows\System32\Windows.UI.Xaml.dll
d2bd749b-d7d6-44e2
Microsoft.Windows.ShellExperienceHost_10.0.14393.187_neutral_neutral_cw5n1hxxxxxxxxx
App

The handle is invalid
```

**Comment by Jaex** [comment]
I have no idea what is ShellExperienceHost. ShareX just adds menu entry using simple registry entry. Not using any shell COM stuffs. So I don't think it is ShareX fault.

**Comment by gaia** [comment]
the crash was immediate. never happened before. maybe you should try to reproduce it first?

**Comment by Jaex** [comment]
How you assume I didn't try already? I selected 10 file and all uploaded successfully.

**Comment by campbeb** [comment]
How is your uploader configured? i.e. What was ShareX trying to do with the csv files?

**Comment by Scrxtchy** [comment]
I've heard of context menu's crashing explorer, but only on the intial opening of them, this is usually done due to a invalid or broken registry entry, or how it interacts with the software.

But never have I seen it crash upon clicking a context menu entry

**Comment by gaia** [comment]
@campbeb uploader has only one setting. send to imgur under an acct.

@Scrxtchy _crashing explorer_ means I see the same result as when I manually run a forced desktop redraw refresh via:

`C:\Windows\System32\taskkill.exe /IM dwm.exe /F`

when I saw this (same as above) happened I knew that explorer had crashed and there it was in the event log.

**Comment by campbeb** [comment]
@gaia From the log file you posted, it looks like it was trying to upload the csv files to Dropbox
```
2016-09-28 23:13:34.918 - Task in queue. Job: FileUpload, Type: File, Host: Dropbox
2016-09-28 23:13:34.944 - Upload started. Filename: 47_filename_20160919.csv, Filepath: D:\Work\47_filename_20160919.csv
2016-09-28 23:13:34.971 - Dropbox configuration is invalid or missing. Please check "Destination settings" window to configure it.

**Comment by gaia** [comment]
@campbeb  yes, i realized that when it is not an img file, it sends to another place. i never used any other upload target nor i have ever configured any other target. HTHs. thanks

**Comment by Jaex** [comment]
I opened fresh ShareX and tried to upload 4x csv file from shell context menu and it just opened destination settings window in ShareX with Dropbox tab selected to tell that Dropbox account is not exists. I just don't see how ShareX can crash explorer. Even if ShareX crash because of error still it must not cause issue in explorer. Because I didn't wrote shell COM code to intercept button click to write my own logic which actually could crash explorer if written wrong.

**Comment by Scrxtchy** [comment]
@Jaex It is indeed an issue with either explorer, or the registry entries of which explorer feeds off.
As mentioned in the OP `The handle is invalid`, the handler of course referring to context menu handlers.
